nntrainer/compiler/slice_realizer.cpp

```diff
@@ -30,20 +30,19 @@ SliceRealizer::realize(const GraphRepresentation &reference) {
     NodeInfo(std::shared_ptr<LayerNode> node) :
       node(node),
       is_visited(false),
-      is_added(false) {}
+      to_be_added(false) {}
     std::shared_ptr<LayerNode> node; /**< set this if not visited */
     bool is_visited;                 /**< set this if visited */
-    bool is_added;                   /**< set this if added */
+    bool to_be_added;                /**< set this if it is to be added */
     std::vector<std::string> children;
-    std::vector<std::string> path;
-    /**< path is the tracing result from start to current node
-      eg) if traversal has started from a -> b -> c -> d.
-      The path has {"a", "b", "c", "d"} */
 
     LayerNode *operator->() { return node.get(); }
   };
 
-  std::unordered_map<std::string, NodeInfo> mp; /// map point
+  /** @note mp has to be ordered map to keep the ordering of the nodes in the
+   * graph */
+  std::map<std::string, NodeInfo> mp; /// map point
```

Comment:
Could you elaborate on this? this will be ordered by name of nodes, plus in this stage should not assume the representation is sorted.

Reply:
If the input_layers is not specified, then the connections are automatically added to the previous layer. Hence ordering is required. This is still buggy as the ordering is by name than the order of insertion. Will update.

Reply:
I recommend you to use unordered_map, this is not related if mp is ordered or not.

We rely on the order of GraphRepresentation to figure out this anyway.

Reply:
yeah, unordered_map is right but the ordering still has to be maintained.
The GraphRepresentation is a vector of layers where ordering of the layers is part of the graph.
Selecting nodes using DFS can change the ordering.
Corresponding code to ensure the ordering is added.